### PR TITLE
Sub: failsafe: subtract times, don't add them

### PR DESCRIPTION
**Pull request by peterbarker**
Avoids integer wrap issues

### CODE REVIEW

**Comment by rmackay9** [comment]
LGTM but really it's @jaxxzer who should give the OK.

**Comment by rmackay9** [comment]
you know.. i'm going to merge this.. it must be correct.

File: ArduSub/failsafe.cpp [hunk 1]
```diff
@@ -314,7 +314,7 @@ void Sub::failsafe_gcs_check()
     uint32_t tnow = AP_HAL::millis();
 
     // Check if we have gotten a GCS heartbeat recently (GCS sysid must match SYSID_MYGCS parameter)
-    if (tnow < failsafe.last_heartbeat_ms + FS_GCS_TIMEOUT_MS) {
+    if (tnow - failsafe.last_heartbeat_ms > FS_GCS_TIMEOUT_MS) {
```
Comment (OXINARF) [inline ArduSub/failsafe.cpp]: This is incorrect, the `<` was incorrectly changed to a `>`.

Comment (rmackay9) [reply ArduSub/failsafe.cpp]: Really?  I've looked at it again and it still appears OK to me..

Comment (OXINARF) [reply ArduSub/failsafe.cpp]: We can look at it as simple math: `failsafe.last_heartbeat_ms` moves to the left, so it gets negative, nothing else changes.

We can also look at it from a logic perspective: this should be true if we are not in a failsafe state. What this was changed to was, if now minus last heartbeat time is bigger than failsafe timeout then it is true... We want the opposite, if now minus last heartbeat time is still smaller than failsafe timeout then it is true.

Comment (rmackay9) [reply ArduSub/failsafe.cpp]: You are right of course.  @peterbarker, I guess you'll PR the fix?  Sorry @OXINARF and thanks for finding this.

**Pull request merged**